src/main/java/com/redhat/labs/omp/socket/SocketAuthenticationFilter.java

```diff
@@ -0,0 +1,92 @@
+package com.redhat.labs.omp.socket;
+
+import java.io.IOException;
+import java.security.Principal;
+
+import javax.inject.Inject;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import javax.servlet.http.HttpServletResponse;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.smallrye.jwt.auth.principal.JWTAuthContextInfo;
+import io.smallrye.jwt.auth.principal.JWTCallerPrincipal;
+import io.smallrye.jwt.auth.principal.JWTCallerPrincipalFactory;
+import io.smallrye.jwt.auth.principal.ParseException;
+
+@WebFilter(urlPatterns = "/engagements/events")
+public class SocketAuthenticationFilter implements Filter {
+
+    public static Logger LOGGER = LoggerFactory.getLogger(SocketAuthenticationFilter.class);
+
+    @Inject
+    private JWTAuthContextInfo authContextInfo;
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+
+        LOGGER.debug("socket authentication filter called with request {} and response {}", request, response);
+
+        HttpServletRequest servletRequest = (HttpServletRequest) request;
+        HttpServletResponse servletResponse = (HttpServletResponse) response;
+
+        // get token from request
+        String token = servletRequest.getParameter("access-token");
```

Comment:
Is the token supposed to be a query param. Normally gets pulled from the header "Authorization: Bearer xxxxx". Is it different for websockets?

Reply:
@mcanoy  Yes, unfortunately it is.  There doesn't seem to be a way from the JavaScript side to pass the token through the headers so Spencer and I agreed on this for now.  It would be nicer in the header, but this was the workaround.  It's also why in the application.properties file I am using a permit all for the endpoint and letting the filter do the authentication.